proto/backend.proto

```diff
@@ -0,0 +1,212 @@
+syntax = "proto3";
+import "google/protobuf/empty.proto";
+package grafana.plugin;
+
+//-----------------------------------------------
+// Common
+//-----------------------------------------------
+
+message PluginConfig {
+  int64 id = 1;
+  int64 orgId = 2;
+  string name = 3;
+  string type = 4;
+  string url = 5;
+  string jsonData = 6;
+  map<string,string> decryptedSecureJsonData = 7;
+}
+
+
+//-----------------------------------------------
+// All Backend Plugins Implement
+//-----------------------------------------------
+
+service BackendPlugin {
+  // Get the plugin status
+  rpc Check(PluginStatusRequest) returns (PluginStatusResponse);
+
+  // HTTP Style request
```

Comment:
It seems like all must implement "Check". The one might really implement only a certain rpc in this whole package. So requiring both Resource and DataQuery doesn't fit with that logic. 

That being said since these are the two ... common? ... methods I am okay with it for now.